torch/utils/benchmark/utils/cpp_jit.py

```diff
@@ -1,15 +1,24 @@
 """JIT C++ strings into executables."""
+import atexit
 import os
+import re
+import shutil
+import textwrap
 import threading
-from typing import List, Optional
+import uuid
+from typing import Any, List, Optional
 
 import torch
-from torch.utils.benchmark.utils._stubs import CallgrindModuleType
+from torch.utils.benchmark.utils._stubs import CallgrindModuleType, TimeitModuleType
 from torch.utils import cpp_extension
 
 
 LOCK = threading.Lock()
 SOURCE_ROOT = os.path.split(os.path.abspath(__file__))[0]
+BUILD_ROOT = os.path.join(
+    torch._appdirs.user_cache_dir(appname="benchmark_utils_jit"),
+    f"build_{uuid.uuid4()}".replace("-", "")
```

Comment:
Calcing the build uuid only once at module import seems pretty skeevy to me...

Reply:
I was about to write a big paragraph on why it's not, but then I thought "that big paragraph should be a comment!".

Reply:
Hmm... well this makes sense, but it raises a new question: do you *trust* `cpp_extension` s recompilation cache? When it comes to benchmarking, I'm not sure I would...

Reply:
I have two safeguards:
1) The `uuid` in the build root, so it won't mix `libtorch` across PyTorch versions.
2) I put the hash of the source in the name before I pass it to `cpp_extension`

So really `cpp_jit.py` guarantees cache correctness; it just has to make sure it doesn't also vary something that `cpp_extension` caches on. (i.e. problems will manifest as excessive recompilation, but never correctness issues.)

Want me to add a note to that effect?